File: app/src/main/java/com/proxy/shadowsocksr/impl/obfs/ObfsChooser.java
package com.proxy.shadowsocksr.impl.obfs;

import com.proxy.shadowsocksr.impl.proto.AbsProtocol;

public final class ObfsChooser
{
    public static AbsObfs getObfs(String protocolName, String rmtIP,
            int tcpMss, String usrParamStr)
    {
        switch (protocolName)
        {
        case "origin":
        default:
            return null;
        }
    }

    public static AbsProtocol getProtocol(String obfsName, String rmtIP,
            int tcpMss, String usrParamStr)
    {
        switch (obfsName)
        {
        case "plain":
        default:
            return null;
        }
    }
}

File: jvmVer/src/main/java/com/proxy/shadowsocksr/impl/plugin/obfs/ObfsChooser.kt
package com.proxy.shadowsocksr.impl.plugin.obfs

import java.util.*

class ObfsChooser
{
    companion object
    {
        fun getObfs(obfsMethod: String, rmtIP: String, rmtPort: Int,
                    tcpMss: Int, usrParamStr: String, shareParam: HashMap<String, Any>): AbsObfs
        {
            when (obfsMethod)
            {
                "http_simple" -> return HttpSimpleObfs(usrParamStr, rmtIP, rmtPort, tcpMss,
                        shareParam)
            //... -> ...
                else -> return SSObfs(usrParamStr, rmtIP, rmtPort, tcpMss,
                        shareParam)//default origin
            }
        }
    }
}
